### ISSUE TITLE
`filter_unhandled_paths` filters out `Mount`ed routes

### ISSUE THREAD

**Issue by axyjo** [stephenhillier/starlette_exporter#21]
Here's a minimal reproduction:

```
from starlette.applications import Starlette
from starlette_exporter import PrometheusMiddleware, handle_metrics
from starlette.routing import Mount, Route
from starlette.responses import PlainTextResponse


async def hello_world(request):
    return PlainTextResponse("hi")


hello_routes = [Route("/world", hello_world)]

app = Starlette(
    debug=True,
    routes=[
        Mount("/hello", routes=hello_routes, name="hello"),
    ],
)
app.add_middleware(PrometheusMiddleware, filter_unhandled_paths=True)
app.add_route("/metrics", handle_metrics)
```

`/metrics` shows requests to `/hello/world` when `filter_unhandled_paths` is False, but not when it's set to True. I'm on starlette_exporter v0.8.2, starlette v0.14.2, python v3.8.10.

**Comment by stephenhillier** [comment]
Thanks for the report and all the info.  I turned your reproduction into a failing test case; I'll let you know once I find a solution.

**Comment by stephenhillier** [comment]
v0.9.0 has been released with the fix from #22. Thanks again and let me know if that fixes the issue with the mounted routes.

**Comment by axyjo** [comment]
Thanks! Will test it out next weekend :)
